### PR TITLE
Added Firebase Auth (signup and login)

### PR DESCRIPTION
**Pull request by diananova**
## AUTHENTICATION:
- /login -> should be working (Firebase Auth)
- /signup -> should be working (Firebase Auth with email and password)
- /users POST -> store user info (name, surname, bio...) on MongoDB (works but integration with Firebase is missing)
- /users GET -> get all users (works, no Firebase Auth), but the user model needs to be changed (name, surname, country, language, bio..). I didn't change it yet because  this way it's easier/quicker to test. 

## TO DO 
- **mongodb’s user id should correspond to the email/id used with Firebase Auth** (not sure how to do this)

## NOTES
- I used Postman (Windows 10) for testing APIs

### CODE REVIEW

File: api/README.md [hunk 1]
```diff
@@ -1 +1,6 @@
-# Express/Node Project
\ No newline at end of file
+# Express/Node Project
+## AUTHENTICATION:
+- /login -> should be working
+- /signup -> Firebase authentication with email and password
+- /users POST -> store user info (name, surname, bio...) on MongoDB
```
Comment (Rita-Okonkwo) [inline api/README.md:5]: Well done! Have you created the route for users?

Comment (diananova) [reply api/README.md:5]: Yep! It's in the api/api-routes.js file.

**Comment by Rita-Okonkwo** [comment]
Hey guys! @diananova @MichaelKMalak I linked the firebase auth to the mongodb database. Please review & confirm if it works

**Review by diananova — COMMENTED** [review]
Hi Rita! Good job, I think we're almost there 😅.

When I run nodemon index, this is what I get:
- mongoose: Cannot specify a custom index on `_id` for model name "user", MongoDB does not allow overwriting the default `_id` index. See http://bit.ly/mongodb-id-index
Db connected successfully
- (node:7016) DeprecationWarning: collection.ensureIndex is deprecated. Use createIndexes instead.
- (node:7016) UnhandledPromiseRejectionWarning: TypeError: **firebaseRef.onAuth is not a function**
  - Also looking at the FIrebase documentation, I can't find onAuth method. Based on what I found on [Stackoverflow](https://stackoverflow.com/questions/55380749/why-do-i-receive-an-error-stating-onauth-is-not-a-function) onAuth method is deprecated.

**Comment by Rita-Okonkwo** [comment]
> Hi Rita! Good job, I think we're almost there 😅.
> 
> When I run nodemon index, this is what I get:
> 
> * mongoose: Cannot specify a custom index on `_id` for model name "user", MongoDB does not allow overwriting the default `_id` index. See http://bit.ly/mongodb-id-index
>   Db connected successfully
> * (node:7016) DeprecationWarning: collection.ensureIndex is deprecated. Use createIndexes instead.
> * (node:7016) UnhandledPromiseRejectionWarning: TypeError: **firebaseRef.onAuth is not a function**
>   
>   * Also looking at the FIrebase documentation, I can't find onAuth method. Based on what I found on [Stackoverflow](https://stackoverflow.com/questions/55380749/why-do-i-receive-an-error-stating-onauth-is-not-a-function) onAuth method is deprecated.

Thanks girl! Alright let’s keep researching. I’m sure we will come up with something soon😄

**Pull request merged**